xarray/core/variable.py

```diff
@@ -492,6 +495,46 @@ def _broadcast_indexes_advanced(self, key):
         key = VectorizedIndexer(variable.data for variable in variables)
         return dims, key
 
+    def _broadcast_indexes_dask_pointwise(self, key):
+        if any(not isinstance(k, (Variable, slice)) for k in key):
```

Comment:
My understandings may be incorrect, 
but why do we restrict indexers only slice and same-shaped Variables?

I think we can apply `broadcast_variables` to indexers (except for slice) if we finally call `moveaxes` in `DaskIndexingAdapter`.